apps/etterlatte-trygdetid/src/main/kotlin/trygdetid/regulering/RegulerTrygdetidService.kt

```diff
@@ -0,0 +1,36 @@
+package no.nav.etterlatte.trygdetid.regulering
+
+import kotlinx.coroutines.runBlocking
+import no.nav.etterlatte.token.Bruker
+import no.nav.etterlatte.trygdetid.Trygdetid
+import no.nav.etterlatte.trygdetid.TrygdetidBeregningService
+import no.nav.etterlatte.trygdetid.TrygdetidRepository
+import no.nav.etterlatte.trygdetid.klienter.BehandlingKlient
+import java.util.*
+
+class RegulerTrygdetidService(
+    private val trygdetidRepository: TrygdetidRepository,
+    private val behandlingKlient: BehandlingKlient,
+    private val beregnTrygdetidService: TrygdetidBeregningService
+) {
+
+    suspend fun regulerTrygdetid(behandlingId: UUID, forrigeBehandlingId: UUID, bruker: Bruker): Trygdetid {
+        val forrigeTrygdetid = trygdetidRepository.hentTrygdetid(forrigeBehandlingId)
+        val forrigeTrygdetidGrunnlag =
+            forrigeTrygdetid?.trygdetidGrunnlag ?: throw Exception("Mangler trygdetid for $forrigeBehandlingId")
+        val regulering = behandlingKlient.hentBehandling(behandlingId, bruker)
+
+        return trygdetidRepository.transaction { tx ->
+
+            trygdetidRepository.opprettTrygdetid(regulering, forrigeTrygdetid.opplysninger, tx)
```

Comment:
Mener å huske Henrik så på en fiks for å kutte i bruk av transactions. Verdt å sjekke ut.

Reply:
Fiksen var å fikse mer i domeneklassen, for så å lagre ned denne (det blir da opp til repository og finne ut hva som er endret etc. Den er ikke merget ennå